src/lambda_handler.py

```diff
@@ -34,54 +164,253 @@ def task_default(event):
     return result
 
 
-def task_cowsay(event) -> dict[str, Union[Optional[str], bool]]:
-    """Generate an output message using the provided information."""
-    result: dict[str, Union[Optional[str], bool]] = {"message": None, "success": True}
+class FileConfig(TypedDict):
+    """Define the type structure of the dictionary in the file_configs variable."""
 
-    character: str = event.get("character", "tux")
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
+    app_regex: re.Pattern[str]
+    description: str
+    filename: str
+    # I'd prefer to define ip_set as Set[Union[IPv4Address, IPv6Address]],
+    # but ipaddress.collapse_addresses() uses the type variable "_N" while
+    # ip_network returns Union[IPv4Network, IPv6network] which causes
+    # the mypy pre-commit hook to throw this error:
+    #   Value of type variable "_N" of "collapse_addresses" cannot be
+    #   "Union[IPv4Network, IPv6Network]"
+    # My solution to this problem is to simply define ip_set as Set[Any].
+    # For a similar issue and discussion, see
+    # https://github.com/python/typeshed/issues/2080
+    ip_set: Set[Any]
+    static_ips: List[str]
+
+
+def validate_event_data(
+    event: Dict[str, Any]
+) -> Tuple[Dict[str, Any], bool, List[str]]:
+    """Validate the event data and return a tuple containing the validated event, a boolean result (True if valid, False if invalid), and a list of error message strings."""
+    result = True
+    errors = []
+
+    # Account ID checks
+    if "account_ids" not in event:
+        errors.append('Missing required key "account_ids" in event.')
+    elif event["account_ids"] == []:
+        errors.append('"account_ids" must be a non-empty list.')
     else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
-        )
-        result["message"] = cowsay.get_output_string(character, contents)
+        account_ids: List[str] = event["account_ids"]
+        try:
+            # Ensure account_ids is a list of strings
+            if type(account_ids) is not list:
+                account_ids = [str(account_ids)]
+            else:
+                account_ids = [str(e) for e in account_ids]
 
-    return result
+            # Verify that each provided AWS account ID is 12 digits
+            invalid_account_ids = []
+            for account_id in account_ids.copy():
+                if not re.match(r"^\d{12}$", account_id):
+                    account_ids.remove(account_id)
```

Comment:
What is the rationale for copying the list and removing elements from the original?

Reply:
I make a copy because removing one or more items from a list while iterating through that same list will skip items in the list.  As with most things in Python, there are other ways to accomplish this- see [here](https://sopython.com/canon/95/removing-items-from-a-list-while-iterating-over-the-list/) for a nice short post on this subject.

I suppose I could have also built up a new list of `valid_account_ids` and just left the original list of account IDs alone.

Did you have a different solution in mind or would you rather see an explanatory comment here?

Reply:
Sorry that was a poorly worded question on my part. I'm familiar with the need to copy a list to remove elements from it while iterating over it in Python. The core of my inquiry was why are we going through the trouble of removing failing account IDs from the list of account IDs.

Reply:
Ah, now I understand.  That logic was left over from when I had previously planned to weed out bad account IDs and continue processing the valid accounts.  Now that we bail out if any errors are found in the event data, it no longer makes sense to remove the bad IDs from the list - updated in https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/a9fa4909a5d0443e60876d14f17ecf264add619e.